### PR TITLE
MAKE-865: set standard input for remote interfaces

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-865

* Add a `StandardInputBytes` field to `CreateOptions` that allows standard input for a process to be serialized over remote interfaces (i.e. REST, RPC).
* (Unrelated) Fix a minor issue where timeout is set  twice - once on context passed to RPC `CreateProcess()`, but again when the process context sets the context in `(*CreateOptions).Resolve()`.

### CODE REVIEW

File: process_test.go [hunk 1]
```diff
@@ -486,6 +488,68 @@ func TestProcessImplementations(t *testing.T) {
 					require.NoError(t, err)
 					assert.True(t, newProc.Complete(ctx))
 				},
+				// "RespawnDoesNotCopyStandardInput": func(ctx context.Context, t *testing.T, opts *CreateOptions, makep ProcessConstructor) {
```
Comment (tychoish) [inline process_test.go]: should we uncomment or remove?

File: process_test.go [hunk 2]
```diff
@@ -554,6 +618,143 @@ func TestProcessImplementations(t *testing.T) {
...
+									Output: output,
+								},
+							}
+							expectedOutput := "foobar"
+							stdin := []byte("echo " + expectedOutput)
+							subTestCase(ctx, t, opts, expectedOutput, stdin, output)
+						})
+					}
+				},
+				// "StandardInputIsSet": func(ctx context.Context, t *testing.T, opts *CreateOptions, makep ProcessConstructor) {
```
Comment (tychoish) [inline process_test.go]: uncomment or remove.

File: process_test.go (hunk 1 above)
Comment (Kimchelly) [reply process_test.go]: Removed. I rewrote this test to consolidate the reusable logic for the standard input tests.

File: process_test.go (hunk 2 above)
Comment (Kimchelly) [reply process_test.go]: Removed.

**Pull request merged**